File: 백준/Silver/2578. 빙고/빙고.py
def findIndex(arr, number):
    for i in range(5):
        for j in range(5):
            if arr[i][j] == number:
                return i, j


answerSheet = [[0, 0, 0, 0, 0] for _ in range(5)]
bingo_cnt = 0
cnt = 0

bingo = []
numbers = []
for _ in range(5):
    row = list(map(int, input().split()))
    bingo.append(row)

for _ in range(5):
    row = list(map(int, input().split()))
    numbers.append(row)

for i in range(5):
    for j in range(5):
        if bingo_cnt >= 3:
            break

        row, col = findIndex(bingo, numbers[i][j])
        answerSheet[row][col] = 1
        cnt+=1


        if all(answerSheet[row][i] == 1 for i in range(5)):
            bingo_cnt+=1

        if all(answerSheet[i][col] == 1 for i in range(5)):
            bingo_cnt+=1
        
        if row == col and all(answerSheet[i][i] == 1 for i in range(5)):
            bingo_cnt+=1
            
        if row+col == 4 and all(answerSheet[4-i][i] == 1 for i in range(5)):
            bingo_cnt+=1

print(cnt)

File: 백준/Silver/2578. 빙고/빙고.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		StringBuilder sb = new StringBuilder();

		int[][] bingo = new int[5][5];
		boolean[][] check = new boolean[5][5];

		String[] index = new String[26];

		int cnt = 0;
		for (int i = 0; i < 5; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			for (int j = 0; j < 5; j++) {
				int value = Integer.parseInt(st.nextToken());
				bingo[i][j] = value;
				index[value] = i + " " + j;
			}
		}
		
		cnt = 0;
		int bingoCnt = 0;
		for (int i = 0; i < 5; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			
			for (int j = 0; j < 5; j++) {
				int num = Integer.parseInt(st.nextToken());
				String[] idx = index[num].split(" ");
				int row = Integer.parseInt(idx[0]);
				int col = Integer.parseInt(idx[1]);
				check[row][col] = true;
				cnt++;
				
				
							
				if(isRowBingo(check, row)) {
					bingoCnt++;
				}
				
				if (isColBingo(check, col)) {
					bingoCnt++;
				}
				
				if (isLeftCrossBingo(check) && row==col) {
					bingoCnt++;
				}
				
				if(isRightCrossBingo(check) && row+col==4) {
					bingoCnt++;
				}
				
				if(bingoCnt>=3) {
					sb.append(cnt).append("\n");
					bingoCnt = Integer.MIN_VALUE;
					break ;
				}	
 			}
			
		}

		bw.write(sb.toString());
		bw.flush();
		bw.close();
		

	}
	
	private static boolean isRowBingo(boolean[][] check, int row) {
		//가로
		for(int i=0; i<5; i++) {
			if(!(check[row][i])) {
				return false;
			}
		}
		return true;
	}
	
	private static boolean isColBingo(boolean[][] check, int col) {
		//세로
		for(int i=0; i<5; i++) {
			if(!(check[i][col])) {
				return false;
			}
		}
		return true;
	}
	
	private static boolean isLeftCrossBingo(boolean[][] check) {
		for (int i=0; i<5; i++) {
			if(!(check[i][i])) {
				return false;
			}
		}
		return true;
	}
	
	private static boolean isRightCrossBingo(boolean[][] check) {
		for (int i=0; i<5; i++) {
			if(!(check[i][4-i])) {
				return false;
			}
		}
		
		return true;
	}


}
